icu4j/main/core/src/test/java/com/ibm/icu/dev/test/calendar/CalendarRegressionTest.java

```diff
@@ -2710,5 +2710,17 @@ public void TestRespectUExtensionFw() { // ICU-22226
                     Calendar.getInstance(Locale.forLanguageTag(localeIds[i])).getFirstDayOfWeek());
         }
     }
+
+    @Test
+    public void TestIslamicUmalquraCalendarSlow() { // ICU-22513
+        Locale loc = new Locale("th@calendar=islamic-umalqura");
+        Calendar cal = Calendar.getInstance(loc);
+        cal.clear();
+        cal.add(Calendar.YEAR, 1229080905);
+        cal.roll(Calendar.WEEK_OF_MONTH, 1499050699);
+        cal.fieldDifference(new Date(0), Calendar.YEAR_WOY);
```

Comment:
Maybe I'm missing something, but shouldn't there be some kind of "expect exception to be thrown" logic in this test?

Reply:
Java use long (64 bits) for those value so it will not throw exception with that setting. The test just make sure it will not be infinity loop or time out.